From 901ed487baeec6d395bd4b6c147aeee6f5104ab5 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:41:05 -0400
Subject: [alsa] hda: Set TLV_DB_SCALE_MUTE bit for cx5051 vmaster

Message-id: <140972823573-420-patchset-jkysela@redhat.com>
Patchwork-id: 91116
O-Subject: [RHEL 7.1 PATCH 419/442] ALSA: hda - Set TLV_DB_SCALE_MUTE bit for cx5051 vmaster
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Conexnat HD-audio driver has a workaround for cx5051 (aka CX20561)
chip to add fake mute controls to each amp (commit 3868137e).  This
implies the minimum-as-mute TLV bit in TLV for each corresponding
control.  Meanwhile we build the virtual master from these, but the
TLV bit is missing, even though the slaves have it.

This patch simply adds the missing TLV_DB_SCALE_MUTE bit for vmaster,
as already done in patch_sigmatel.c.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 61074c1a2d798be90e149a9ed4ce0e3596c00a38)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_conexant.c b/sound/pci/hda/patch_conexant.c
index 7627a69..6f2fa83 100644
--- a/sound/pci/hda/patch_conexant.c
+++ b/sound/pci/hda/patch_conexant.c
@@ -26,6 +26,7 @@
 #include <linux/module.h>
 #include <sound/core.h>
 #include <sound/jack.h>
+#include <sound/tlv.h>
 
 #include "hda_codec.h"
 #include "hda_local.h"
@@ -859,6 +860,11 @@ static int patch_conexant_auto(struct hda_codec *codec)
  if (err < 0)
   goto error;
 
+ if (codec->vendor_id == 0x14f15051) {
+  /* minimum value is actually mute */
+  spec->gen.vmaster_tlv[3] |= TLV_DB_SCALE_MUTE;
+ }
+
  codec->patch_ops = cx_auto_patch_ops;
 
  /* Some laptops with Conexant chips show stalls in S3 resume,
-- 
1.7.1